src/main/java/com/meta/cp4m/metrics/CAPIMetricsCollector.java

```diff
@@ -0,0 +1,148 @@
+    private final ConcurrentHashMap<EventRecord, Long> messageCounts = new ConcurrentHashMap<>();
+    private static final CAPIMetricsCollector HANDLER = new CAPIMetricsCollector();
+    public CAPIMetricsCollector() {
+        ScheduledExecutorService threadPool = Executors.newScheduledThreadPool(1);
+        threadPool.scheduleAtFixedRate(this::sendAllMetrics, 1, 5, TimeUnit.MINUTES);
+    }
+    private void sendSingleMetric() {
+
+        String ACCESS_TOKEN = "EAAKxnuilYZAQBO2dbiWA3r8z2FMzdmU2ZAfS3idlE1mzRXbZBJWDnZBmJNihs1CYz8GRZBl83jaErtsckFr8M88tWHUafREOEvFfDsq9vvSByks57y3sv5DZCa5IUqCSSjDwR4OJS2QkBogSvZAoCyO0KDJn69RzZBrQZCPeCk87PvfRf3yJP7gYQdL6H1dSD5BjSZCMN2EmgVM5aELrhGh3EZD";
+        String PIXEL_ID = "1452308421990109";
+        APIContext context = new APIContext(ACCESS_TOKEN);
+
+        List<Event> events = new ArrayList<>();
+
+        UserData userData_0 = new UserData()
+                .clientIpAddress("207.96.79.161");
+
+
+            CustomData customData_0 = new CustomData();
+            HashMap<String, String> customProperties_0 = new HashMap<String, String>();
+            customProperties_0.put("plugin", "openai");
+            customProperties_0.put("messenger", "whatsapp");
+            customProperties_0.put("business_contact_ID", "3333333333");
+            customData_0.customProperties(customProperties_0);
+
+            Event event_0 = new Event()
+                    .eventName("WAReceivedJSON")
+                    .eventTime(1717122946L)
+                    .userData(userData_0)
+                    .customData(customData_0)
+                    .actionSource(ActionSource.valueOf("website"));
+
+            events.add(event_0);
+            EventRequest eventRequest = new EventRequest(PIXEL_ID, context)
+                    .data(events);
+
+        try {
+            EventResponse response = eventRequest.execute();
+            System.out.printf("Standard API response : %s ", response);
+        } catch (APIException e) {
+            e.printStackTrace();
+        }
+    }
+
+    private void sendAllMetrics() {
+        String ACCESS_TOKEN = "EAAKxnuilYZAQBO2dbiWA3r8z2FMzdmU2ZAfS3idlE1mzRXbZBJWDnZBmJNihs1CYz8GRZBl83jaErtsckFr8M88tWHUafREOEvFfDsq9vvSByks57y3sv5DZCa5IUqCSSjDwR4OJS2QkBogSvZAoCyO0KDJn69RzZBrQZCPeCk87PvfRf3yJP7gYQdL6H1dSD5BjSZCMN2EmgVM5aELrhGh3EZD";
+        String PIXEL_ID = "1452308421990109";
+
+        APIContext context = new APIContext(ACCESS_TOKEN);
+
+        List<Event> events = new ArrayList<>();
+
+        messageCounts.forEach((event, count) -> {
+            CustomData customData_0 = new CustomData();
+            HashMap<String, String> customProperties_0 = new HashMap<String, String>();
+
+
+            UserData userData_0 = new UserData()
+                    .clientIpAddress("207.96.79.161")
+                    .clientUserAgent("cp4m")
```

Comment:
hardcoding

Reply:
Do we actually need to get the correct IP address for the advertiser using it? since we're not using this field..